js/src/framework.test.ts

```diff
@@ -555,3 +555,124 @@ test("Eval with noSendLogs: true runs locally without creating experiment", asyn
   await memoryLogger.flush();
   expect(await memoryLogger.drain()).toHaveLength(0);
 });
+
+test("tags can be appended and logged to root span", async () => {
+  await _exportsForTestingOnly.simulateLoginForTests();
+  const memoryLogger = _exportsForTestingOnly.useTestBackgroundLogger();
+  const experiment =
+    _exportsForTestingOnly.initTestExperiment("js-tags-append");
+
+  const initialTags = ["cookies n cream"];
+  const appendedTags = ["chocolate", "vanilla", "strawberry"];
+  const expectedTags = [
+    "cookies n cream",
+    "chocolate",
+    "vanilla",
+    "strawberry",
+  ];
+
+  const result = await runEvaluator(
+    experiment,
+    {
+      projectName: "proj",
+      evalName: "js-tags-append",
+      data: [{ input: "hello", expected: "hello world", tags: initialTags }],
+      task: (input, hooks) => {
+        for (const t of appendedTags) hooks.tags.push(t);
+        return input;
+      },
+      scores: [() => ({ name: "simple_scorer", score: 0.8 })],
+      summarizeScores: false,
+    },
+    new NoopProgressReporter(),
+    [],
+    undefined,
+  );
+  expect(result.results[0].tags).toEqual(expectedTags);
+
+  await memoryLogger.flush();
+  const logs = await memoryLogger.drain();
+  const rootSpans = logs.filter((l: any) => !l["span_parents"]);
+  expect(rootSpans).toHaveLength(1);
+  expect((rootSpans[0] as any).tags).toEqual(expectedTags);
+});
+
+test.each([
+  {
+    title: "empty list returns undefined for tags",
+    providedTags: [],
+    expectedTags: undefined,
+  },
+  {
+    title: "tags can be set to a list",
+    providedTags: ["chocolate", "vanilla", "strawberry"],
+    expectedTags: ["chocolate", "vanilla", "strawberry"],
+  },
+])("$title", async ({ providedTags, expectedTags }) => {
+  await _exportsForTestingOnly.simulateLoginForTests();
+  const memoryLogger = _exportsForTestingOnly.useTestBackgroundLogger();
+  const experiment = _exportsForTestingOnly.initTestExperiment("js-tags-list");
+
+  const result = await runEvaluator(
+    experiment,
+    {
+      projectName: "proj",
+      evalName: "js-tags-list",
+      data: [{ input: "hello", expected: "hello world" }],
+      task: (input, hooks) => {
+        hooks.tags = providedTags;
+        return input;
+      },
+      scores: [() => ({ name: "simple_scorer", score: 0.8 })],
+      summarizeScores: false,
+    },
+    new NoopProgressReporter(),
+    [],
+    undefined,
+  );
+  expect(result.results[0].tags).toEqual(expectedTags);
+
+  await memoryLogger.flush();
+  const logs = await memoryLogger.drain();
+  const rootSpans = logs.filter((l: any) => !l["span_parents"]);
+  expect(rootSpans).toHaveLength(1);
+  expect((rootSpans[0] as any).tags).toEqual(expectedTags);
+});
+
+test("tags are persisted with a failing scorer", async () => {
```

Comment:
similar to the comment on the python tests - should we add one more test where no initial tags are set on the eval case and we dont do anything with the tags hook?
that should behave the same with and without this change so it's a good regression test